libsolutil/Keccak256.cpp

```diff
@@ -69,25 +69,25 @@ static uint64_t const RC[24] = \
 #define REPEAT6(e) e e e e e e
 #define REPEAT24(e) REPEAT6(e e e e)
 #define REPEAT5(e) e e e e e
-#define FOR5(v, s, e) \
+#define FOR5(type, v, s, e) \
 	v = 0;            \
-	REPEAT5(e; v += s;)
+	REPEAT5(e; v = static_cast<type>(v + s);)
 
 /*** Keccak-f[1600] ***/
 static inline void keccakf(void* state) {
-	uint64_t* a = (uint64_t*)state;
+	auto* a = (uint64_t*)state;
 	uint64_t b[5] = {0};
 
 	for (int i = 0; i < 24; i++)
 	{
 		uint8_t x, y;
 		// Theta
-		FOR5(x, 1,
+		FOR5(uint8_t, x, 1,
```

Comment:
would `FOR5(x, static_cast<uint8_t>(1))` work?

Reply:
I think `FOR5(x, 1u...)` might work too.

Reply:
Tried both before converting whole expression itself
I think that `v + s` get promoted to int and that is why cast of whole expression is needed